crates/bevy_reflect/src/reflect.rs

```diff
@@ -73,9 +73,6 @@ pub enum ReflectOwned {
 /// [derive macro]: bevy_reflect_derive::Reflect
 /// [crate-level documentation]: crate
 pub trait Reflect: DynamicTypePath + Any + Send + Sync {
-    /// Returns the [type name][std::any::type_name] of the underlying type.
-    fn type_name(&self) -> &str;
```

Comment:
iiuc, its almost functionally equivalent to a rename `Reflect::type_name` to `DynamicTypePath::reflect_type_path`; `DynamicTypePath` doesn't even need to be in scope to call `my_reflect.reflect_type_path()`, but i suppose we could give it a deprecated default impl that calls `reflect_type_path`.

Reply:
Yeah a deprecation would be good if we can make it work (I bring it up because we should probably start using it more where we can). If not, or if it's too much trouble, then we can just skip it